tutorials/shading/migrating_to_godot_shader_language.rst

```diff
@@ -0,0 +1,211 @@
+.. _doc_migrating_to_godot_shader_language:
+
+Migrating to Godot shading language
+===================================
+
+Introduction
+------------
+
+This document will explain the differences between Godot's shading language
+and glsl as well as give practical advice on how to migrate shaders from other
+sources, such as Shadertoy and The Book of Shaders, into Godot shaders.
+
+For detailed information on Godot's shading language please refer to the :ref:`Shading Language <doc_shading_language>`
+reference.
+
+Glsl
+----
+
+Godot uses a shading language based on glsl, with the addition of a few quality of life features.
+Accordingly, most features available in glsl are available in Godot's shading language.
+
+Shader Programs
+^^^^^^^^^^^^^^^
+
+In glsl each shader uses a seperate program. So you have one program for the vertex shader and  one 
+for the fragment shader. In godot you have a single shader that contains a ``vertex`` and/or a 
+``fragment`` function. If you only choose to write one, Godot will use a default  for the other. 
```

Comment:
Remove extra space between "default" and "for"

Explain "use a default" - what does this mean for me if I don't write one?